src/Utilities/Compiler/Extensions/ISymbolExtensions.cs

```diff
@@ -649,6 +649,31 @@ public static IEnumerable<AttributeData> GetAttributes(this ISymbol symbol, para
             return symbol.GetAttributes(attributesToMatch: attributeTypesToMatch);
         }
 
+        #region "Overloads to avoid array allocations"
+        public static IEnumerable<AttributeData> GetAttributes(this ISymbol symbol, INamedTypeSymbol? attributeTypeToMatch1)
+        {
+            foreach (var attribute in symbol.GetAttributes())
+            {
+                if (SymbolEqualityComparer.Default.Equals(attribute.AttributeClass, attributeTypeToMatch1))
+                {
+                    yield return attribute;
+                }
+            }
+        }
+
+        public static IEnumerable<AttributeData> GetAttributes(this ISymbol symbol, INamedTypeSymbol? attributeTypeToMatch1, INamedTypeSymbol? attributeTypeToMatch2)
```

Comment:
Note: This is called for every named type, method, property, field, and event in the compilation. So using the `params` overload was likely inefficient.

An alternative is that the analyzer would pre-allocate a single array at compilation start, but I went with these overloads so that other analyzers could benefit as well.